app/models/candidate_interface/personal_details_form.rb

```diff
@@ -8,8 +8,37 @@ class PersonalDetailsForm
                   :english_main_language,
                   :english_language_details, :other_language_details
 
+    validates :first_name, :last_name, :english_main_language,
+              :first_nationality,
+              presence: true
+
+    validates :first_name, :last_name,
+              length: { maximum: 100 }
+
+    validate :date_of_birth_cannot_be_in_the_future
+
+    validates :first_nationality, :second_nationality,
+              inclusion: { in: NATIONALITIES, allow_blank: true }
+
+    validates :english_language_details, :other_language_details,
+              word_count: { maximum: 200 }
+
     def name
       "#{first_name} #{last_name}"
     end
+
+    def date_of_birth
+      Date.new(*[year, month, day].map(&:to_i))
+    rescue ArgumentError, NoMethodError
+      nil
+    end
+
+    def date_of_birth_cannot_be_in_the_future
+      raise 'Invalid date' if [year, month, day].any?(&:blank?)
+
+      errors.add(:date_of_birth, I18n.t('activemodel.errors.models.candidate_interface/personal_details_form.attributes.date_of_birth.future')) if date_of_birth > Date.today
+    rescue RuntimeError, NoMethodError
+      errors.add(:date_of_birth, I18n.t('activemodel.errors.models.candidate_interface/personal_details_form.attributes.date_of_birth.invalid'))
```

Comment:
I feel sure there's a way to do this without writing out the fully-qualified path of the translation. 🐇

Reply:
Maybe pass `:invalid` in the second place? https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-add

Reply:
Nice, we'll give that a go!